### DISCUSSION TITLE
Contabilizzare consumi

### DISCUSSION THREAD

**Discussion by AE-PAVAN** [indomus/forum#2415] [GENERALE - Discussioni generiche]
Ciao a tutti, mi chiamo Antonio. Non è molto che mi sono affacciato al mondo di Home Assistant e una delle cose che più mi ha intrigato è la lettura di un articolo sui ["contabilizzato dei consumi](https://indomus.it/progetti/configurare-dei-contabilizzatori-di-consumo-energetico-su-home-assistant-v2/)". Nei commenti ho letto un post che rimandava ad una implementazione delle tariffe e ho "copiato" quando messo a disposizione. Ho sostituito il codice con i miei sensori cercando di capire anche l'operatività (questo per non fare un semplice copia incolla, ma apprendere come poi applicare in altri contesti). Purtroppo, facendo una verifica della configurazione, Home Assistant mi segnala un errore legato al template sensore (messo nel file sensors.yaml). Lo incollo qui sotto sperando di riuscire a fargli mantenere  la corretta intendazione.

```
- platform: template
  sensors:
    name: "Costo energia"
    unit_of_measurement: '€'
    device_class: monetary
    state_class: total_increasing
  # commercializzazione al mese: 7.471517
    # quota fissa al mese: 1.72
    # quota potenza impegnato al mese a Kw (8): 1.71*8
    # costo materia energia in F1 a kwh: XXXXX
    # costo materia energia in F2 a kwh: YYYYY
    # costo materia energia in F3 a kwh: ZZZZZ
    # dispacciamento a kwh: 0.017258
    # quota energia a kwh: 0.00848
    # quota energia -meccanismi perequativi- a kwh: 0.000095
    # accisa su materia energia a kwh: 0.0227
    # IVA al 10% sul totale: 1.1
    state: > 
      " {{ ((
        7.471517 / states('sensor.giorni_nel_mese_corrente') |float(0) * now().day |float(0) +
        1.72 / states('sensor.giorni_nel_mese_corrente') |float(0) * now().day |float(0) +
        1.71*8 / states('sensor.giorni_nel_mese_corrente') |float(0) * now().day |float(0) +
        states('sensor.consumo_mensile_f1') |float(0) * 997 +
        states('sensor.consumo_mensile_f2') |float(0) * 998 +
        states('sensor.consumo_mensile_f3') |float(0) * 999 +
        (states('sensor.consumo_mensile_f1') |float(0) + states('sensor.consumo_mensile_f2') |float(0) + states('sensor.consumo_mensile_f3') |float(0)) * 0.017258 +
        (states('sensor.consumo_mensile_f1') |float(0) + states('sensor.consumo_mensile_f2') |float(0) + states('sensor.consumo_mensile_f3') |float(0)) * 0.00848 +
        (states('sensor.consumo_mensile_f1') |float(0) + states('sensor.consumo_mensile_f2') |float(0) + states('sensor.consumo_mensile_f3') |float(0)) * 0.000095 +
        (states('sensor.consumo_mensile_f1') |float(0) + states('sensor.consumo_mensile_f2') |float(0) + states('sensor.consumo_mensile_f3') |float(0)) * 0.0227) *
        1.1) |round(2) 
      }}
```

Rispetto all'originale ho apportato le seguenti modifiche:
1) ho cambiato i nomi dei sensori per riprendere i miei;
2) ho cambiato (temporaneamente) i valori XXXX, YYYY e ZZZZ in modo totalmente "arbitrario" in quanto devo andare a vedere sul sito i prezzi delle tariffe.
Mi sono tenuto i commenti per ricostruire gli altri prezzi: al momento ho solo verificato l'accisa e l'IVA, gli altri devo ancora vedere come recuperarli: se vi è anche possibile dirmi se li posso estrapolare dalla bolletta...
Come ultima cosa aggiungo che ho provato a copiare il template nei "Modelli" e mi espone correttamente il risultato dei calcoli.
Incollo qui sotto l'errore che viene visualizzato quando faccio la verifica della configurazione:

```
Invalid config for [sensor.template]: expected dictionary for dictionary value @ data['sensors']['device_class']. Got 'monetary' 
expected dictionary for dictionary value @ data['sensors']['name']. Got 'Costo energia' expected dictionary for dictionary value @ data['sensors']['state']. Got '" {{ ((\n 7.471517 / states(\'sensor.giorni_nel_mese_corrente\') |float(0) * now().day |float(0) +\n 1.72 / states(\'sensor.giorni_nel_mese_corrente\') |float(0) * now().day |float(0) +\n 1.71*8 / states(\'sensor.giorni_nel_mese_corrente\') |float(0) * now().day |float(0) +\n states(\'sensor.consumo_mensile_f1\') |float(0) * 997 +\n states(\'sensor.consumo_mensile_f2\') |float(0) * 998 +\n states(\'sensor.consumo_mensile_f3\') |float(0) * 999 +\n (states(\'sensor.consumo_mensile_f1\')... expected dictionary for dictionary value @ data['sensors']['state_class']. Got 'total_increasing' expected dictionary for dictionary value @ data['sensors']['unit_of_measurement']. Got '€'. (See ?, line ?).
```

Un pò lungo, ma spero di aver dato indicazioni sufficienti perché possiate aiutarmi.
Grazie.
Antonio

**Comment by maxcanna** [comment]
Ciao Antonio, hai fatto un po' di confusione nella definizione del tuo template sensor. Hai mischiato nuova e vecchia versione (basata su `platform`).
Ti consiglio di leggere la documentazione ufficiale e adattare il tuo sensore https://www.home-assistant.io/integrations/template/
